### PR TITLE
hashi_vault: Perform a LIST request when secret path ends with a slash

### PR DESCRIPTION
**Pull request by Normo**
##### SUMMARY
Backwards-compatible, minimal adaption of 'hashi_vault' lookup script to also perform a LIST request, which is an essential part of HashiCorp Vault API.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
- hashi_vault lookup plugin

##### ADDITIONAL INFORMATION

Example Playbook (you should set environment variables VAULT_ADDR and VAULT_TOKEN to access your running HashiCorp Vault instance):
```
- name: Accessing a path not ending in a slash (this works!)
  debug:
    msg:
      - "{{ lookup('hashi_vault', 'secret=secrets/some/path/some_secret1') }}"

- name: Accessing a path ending with a slash should return a list of subsequent secrets. (This does not work, yet!)
  loop: "{{ lookup('hashi_vault', 'secret=secrets/some/path/:keys') }}"
  debug:
    msg: "{{ item }}"
```

```
Before change:

fatal: [example.com]: FAILED! => {"msg": "An unhandled exception occurred while running the lookup plugin 'hashi_vault'. Error was a <class 'ansible.errors.AnsibleError'>, original message: The secret secrets/some/path/ doesn't seem to exist for hashi_vault lookup"}


After change:

ok: [example.com] => {
    "msg": [
        "some_secret1",
        "some_secret2",
        "..."
    ]
}
```

### CODE REVIEW

**Comment by Normo** [comment]
ready_for_review

**Comment by Normo** [comment]
bot_status

File: lib/ansible/plugins/lookup/hashi_vault.py [hunk 1]
```diff
@@ -68,12 +68,22 @@
...
+
+- name: Return one field with specific name within a secret.
+  debug:
     msg: "{{ lookup('hashi_vault', 'secret=secret/hello:value token=c975b780-d1be-8016-866b-01d0f9b688a5 url=http://myvault:8200')}}"
 
-- name: Return all secrets from a path
+- name: Return a list of all secrets under a given path (which must end in a slash)
   debug:
-    msg: "{{ lookup('hashi_vault', 'secret=secret/hello token=c975b780-d1be-8016-866b-01d0f9b688a5 url=http://myvault:8200')}}"
+    msg: "{{ lookup('hashi_vault', 'secret=secret/:keys token=c975b780-d1be-8016-866b-01d0f9b688a5 url=http://myvault:8200') }}"
```
Comment (loksonarius) [inline lib/ansible/plugins/lookup/hashi_vault.py:81]: I'm not sure I understand how a list operation is semantically compatible with a `:keys` at the end.
I'd figure the list operation would not accept a field to list for each secret under a path.

**Comment by timcrawf** [comment]
Is there anything I can do to help this get merged in quicker?

**Comment by Kassec** [comment]
Would be nice feature to have.

**Comment by tsuna** [comment]
Has this PR effectively been canned? Would be useful to have indeed.

**Comment by mkrizek** [comment]
@tsuna Feel free to move it to https://github.com/ansible-collections/community.general where the plugin lives now. Thanks!

**Pull request closed without merging**